### ISSUE TITLE
Worker promotion bug in the latest version

### ISSUE THREAD

**Issue by DanVercetti** [yairm210/Unciv#2224]
![Screenshot_2020-03-22-05-59-46](https://user-images.githubusercontent.com/61697783/77237829-6ba80b00-6c06-11ea-80eb-71f984977718.png)
![Screenshot_2020-03-22-05-59-55](https://user-images.githubusercontent.com/61697783/77237836-78c4fa00-6c06-11ea-8e41-ef414079bb49.png) 

[H4sIAAAAAAAAAOy.txt](https://github.com/yairm210/Unciv/files/4364377/H4sIAAAAAAAAAOy.txt)

The bug enables me to somehow promote two of my workers when I view them in the "Units" tab. Take care and stay away from the virus devs!

**Comment by rh-github-2015** [comment]
I did that star and suspect right away that another thing I coded would have prevented that. Got a save?

Or check whether your save built those workers pre - #2192: That code prevented workers getting XP from barracks et al under the hood but did (for simplicity) not remove XP from already existing ones.

I'll look into the code when I have the time (UnitPromotions.canBePromoted() clearly shows you are right under abovementioned circumstances but I'll have to think a little whether hardening there with a isCivilian check is the best solution)
